src/xb-self-test.c

```diff
@@ -9,6 +9,7 @@
 #include <gio/gio.h>
 #include <locale.h>
 
+#include "xb-arena.h"
```

Comment:
Looking at the existing unit tests made me think it might be a good idea to have some explicit unit tests for `XbArena`, just to check corner case behaviour which would cause hard-to-debug problems if it went wrong.

For example:
 - Zero-sized allocations always return `NULL`
 - Returned allocated pointers are always appropriately aligned for the requested size
 - Corner cases with the utility functions (like `strdup` of `NULL` or of an empty string)